sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/QuantileSummaries.scala

```diff
@@ -41,7 +41,7 @@ import org.apache.spark.sql.catalyst.util.QuantileSummaries.Stats
  * @param count the count of all the elements *inserted in the sampled buffer*
  *              (excluding the head buffer)
  */
-class QuantileSummaries(
+case class QuantileSummaries(
```

Comment:
Make it as a case class, so that we can use the expression encoder to serialize it.